### ISSUE TITLE
Discussion between me and FelixBenning

### ISSUE THREAD

**Issue by dexterlemmer** [dexterlemmer/impl4#2]
@FelixBenning . I heartily agree that we should get this discussion off of this issue. We could try to talk in person. However, there's no discord channel on Rust discord yet that are on topic. We can may be friend each other and talk privately? I don't currently use any other suitable social media and I don't intend to either. Oh. We could move the discussion to a new issue in this repo or on one of my or your repos. So here's the new issue, if we decide to go that route.

Continued from https://github.com/rust-dataframe/discussion/issues/6#issuecomment-642315018.

**Comment by dexterlemmer** [comment]
I wonder whether we even need to bother continuing this discussion at all. It may be insightful to ask ourselves what we want with this discussion in the first place.

### What exactly is your goal with this discussion?
Here's what it seems to me your goal may be:
* To convince me I can encode any type I need with enums+multiindexes?
    * I know you can. In fact, I know you can do it with nothing but a `Vec<u8>`! So there. I'm convinced. I also don't care. There are other concerns in actual projects than what is possible and even what is commonly done.
* To convince me to use an enums-based dataframe?
    * No way!!! Enum variants are dynamically typed! Enums are also wasteful in memory. Just the memory waste would likely double computation time on my current project[^1]! Enums are also unergonomic for some use cases, requiring pattern matching and re-implimentation of traits all over the place while the compiler could've done the job for you if you just used a statically typed type!
* To tell me what you plan to work on yourself?
    * Nice. Say so, then. Then have fun. I'll rather use [arrow](https://crates.io/crates/arrow) or [agnes)(https://github.com/agnes-rs/agnes) or [nevi-me's rust-dataframe](https://github.com/nevi-me/rust-dataframe) in stead while working on my own crate.

### What exactly is my goal with this discussion?
* To convince you that an enums-based dataframe is not all that great for all use cases?
    * Well, yeah! But it works for you. And most of the time it'll be sufficient for me as well. And there are already alternatives available. Frankly, just looking at the other projects that already exist, I think arrow as back-end combined with higher level dataframe API crates built on top of arrow is the way things are going and it also looks like perhaps it's a good way for things to go.
* To figure out my impl4:
    * Yeah well. I think I've got the gist of it already. And here is not the place for it, any way.
* To come up with a better way to describe the different approaches (impl1/2/3/4) and the pros/cons of them.
    * Then may be I should do some coding and some experience using them. :wink: 

### Conclusion:
It seems we're all out of things to say any way. Unless, your goal with the discussion is something I didn't list or you can teach me where I'm wrong about Rust. (For example that I'm wrong about enums.) But that I'll probably figure out myself pretty soon if I actually get to spend some time coding in Rust.

[^1] Ofc, I can't know enums would double my computation time. Rust is very different from Python. However, I have strong reason to suspect it would.

**Comment by dexterlemmer** [comment]
OK. Actually we may have something to talk about. It might make sense for us to decide on an abridged and reorganized version of our discussion we could put back on the original issue. Then delete our long posts there? I feel really bad about how we clogged it up.

**Comment by FelixBenning** [comment]
@dexterlemmer I agree that simplifying our discussion would be desirable. But in order to do that we first need to understand each others proposals. I do not understand yours, so I would greatly appreciate if you could take the time and explain it to me in person. And from your memory concerns you probably assumed that I wanted to enum every field, which is not my intention. The enum would be for the columns, that would only be one additional usize per column. Which will hardly make a dent, memory alignment buffers are going to eat up more memory in virtually all cases. So from these concerns which do not match the actual idea, it seems obvious to me, that we both have not understood each other

So what would you use for communication? Skype? Hangouts? Zoom? Discord? I guess Mumble might work too, but there you do not have screen sharing which might be helpful in our case.

**Comment by dexterlemmer** [comment]
I'd prefer Discord.
I guess we could use Hangouts as well. (I forgot that I do use it -- occasionally.) But I am not really very familiar with it.
That said. I don't see how screen sharing would help. I guess we could start showing Python or R code to each other or I can re-familiarize myself with Rust sufficiently to actually life code what I need/want to show you. However, Python/R is too different from Rust for much more than showing my use cases. And I won't be ready for Rust by this evening.
Voice might help, but I have Asperger's disease. Talking or showing is harder for me than writing. Live texting would have considerable advantages over github issues. And live texting + voice would probably be even better. Adding screen sharing -- with which I'm not well practiced -- might in this case actually make it harder -- at least for me -- to make myself understood. But yea. With discord or hangouts, we get that option in case it helps or in case you can explain something better to me even if not the other way around.

**Comment by FelixBenning** [comment]
`Felix Benning#4222` on discord. And I am not really after working code. I just want to understand your approach, so I thought we could write some pseudo code while talking. Talking + writing is more bandwidth than just writing ;-). So yeah I had live texting + voice in mind as well. That said, I have just realized, that I had a time zone mixup and thus have a meeting today at 13:00 EST and not CEST, which happens to be 19:00 CEST. Soo... could we reschedule?

**Comment by dexterlemmer** [comment]
Yes we could reschedule. Any time between 17:00 CEST and 20:00 CEST most days would work for me. Or on Saturday.
I'm PhireSpere#2311 on Discord.

**Comment by FelixBenning** [comment]
okay, then I would suggest 18:00 CEST Saturday?
